doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,526 @@
+  active control's CWD.
+  * We don't need to have the control raise an event when the CWD changes - we
+    can lazy-load these actions when a UI element that requires it is first
+    invoked.
+* The Command Palette is trickier, since it binds directly to the action map.
+  We'd need to be able to freely modify that map at runtime, which might be
+  prohibitively annoying. <!-- * If we want these actions to show up in the
+  Command Palette, we'll need to:
+  * We'll stash these actions in the action map as they're loaded. (in
+    `Terminal.Settings.Model`)
+  * We'll need to be able to dynamically remove them at runtime from the map (in
+    `Terminal.Settings.Model`) -->
+* If we find multiple `.wt.json` files in the ancestors of the CWD (e.g. for
+  `c:\a\b\c\d\`, there's a `c:\a\.wt.json` and a `c:\a\b\c\.wt.json`), then
+  we'll add each one separately to the map of paths->directories. When
+  requesting the actual actions for `c:\a\b\c\d\`, we'll layer the ones from
+  `c:\a\` before the ones from `c:\a\b\c`, so that deeper descendants take
+  precedence.
+
+### Saving snippets from the commandline
+
+_This has already been prototyped in [#16513]_
+
+Users should be able to save commands as snippets directly from the commandline.
+Consider: you've just run the command that worked the way you need it to. You
+shouldn't have to open the settings to then separately copy-paste the command in
+to save it. It should be as easy as <kbd>Up</kbd>, <kbd>Home</kbd>, `wt save `,
+<kbd>Enter</kbd>.
+
+The exact syntax as follows:
+
+This will be powered by a `saveSnippet` action behind the scenes. After running
+this command, a toast will be presented to the user to indicate success/failure.
+
+#### `save` subcommand
+
+`save [--name,-n name][--description,-d description][-- commandline]`
+
+Saves a given commandline as a sendInput action to the Terminal settings file.
+This will immediately write the Terminal settings file.
+
+**Parameters**:
+* `--name,-n name`: The name to assign to the `name` parameter of the saved
+  command. If omitted, then the parameter will be left blank, and the command
+  will use the auto-generated "Send input:..." name in menus.
+* `--description,-d`: The description to optionally assign to the command.
+* `commandline`: The commandline to save as the `input` of the `sendInput` action.
+
+If the `save` subcommand is ran without any other subcommands, the Terminal will
+imply the `-w 0` arguments, to attempt to send this action to the current
+Terminal window. (unless of course, `-w` was manually provided on the
+commandline). When run with other subcommands, then the action will just be ran
+in the same window as all the other subcommands.
+
+> [!NOTE] In other team discussions, we've considered initially merging this
+> subcommand as `x-save`, where `x-` implies "experimental". We may want to use
+> that to merge [#16513] while we wait for this spec to be approved.

```

Comment:
Personally, I'd prefer we just call it `save` from the start or we wait for the spec to merge. We'd probably have to keep support for `x-save` forever for legacy purposes and that just sounds icky, but that's just my 2 cents.

Reply:
> We'd probably have to keep support for x-save forever

ah, the whole point of `x-` is "this is experimental and will _definitely_ go away in a future build"

Reply:
Also, I've got a mind to merge #16513 while this spec is in review.